common/calculate-swap3.ts

```diff
@@ -0,0 +1,260 @@
+type Swap3LiquidityPosition = {
+  // TODO: Record who added this stuff?
+
+  // Not sure if this is needed; maybe YES and NO left
+  // amount: number // M$ quantity
+
+  // For now, only support YES and NO outcome tokens
+  // TODO: replace with Outcome
+  // Hm, is this...
+  // 1. Number of shares left in this particular pool?
+  // 2. Fixed at injection time?
+  pool: { YES: number; NO: number }
+
+  // Uniswap uses 0.01, 0.003, 0.0005. Let's stick with 0.003 for now.
+  // fee: number
+
+  // Min/max is expressed as a odds ratio of cost of YES to cost of NO
+  // E.g. ratio of 1 = 1:1 = 50%; ratio of 3 = 3:1 = 75%
+  // minRatio: number
+  // maxRatio: number
+  minTick: number

```

Comment:
Might want to create a design doc for the specific data structures you plan to use. 

I think it would probably be easier to just store the probability range here. It's easy to compute the tick from the probability, and this version is not easily human parseable. (It also isn't robust to changing the tick size in the future.)